ci-operator/step-registry/gather/alibabacloud-console/gather-alibabacloud-console-commands.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+trap 'CHILDREN=$(jobs -p); if test -n "${CHILDREN}"; then kill ${CHILDREN} && wait; fi' TERM
+
+
+
+if test -f "${KUBECONFIG}"
+then
+	oc --request-timeout=5s get nodes -o jsonpath --template '{range .items[*]}{.spec.providerID}{"\n"}{end}' | sed 's|.*/||' > "${TMPDIR}/node-provider-IDs.txt" &
+	wait "$!"
+
+	oc --request-timeout=5s -n openshift-machine-api get machines -o jsonpath --template '{range .items[*]}{.spec.providerID}{"\n"}{end}' | sed 's|.*/||' >> "${TMPDIR}/node-provider-IDs.txt" &
+	wait "$!"
+else
+	echo "No kubeconfig; skipping providerID extraction."
+fi
+
+if test -f "${SHARED_DIR}/alibaba-instance-ids.txt" && test -s "${SHARED_DIR}/alibaba-instance-ids.txt"
```

Comment:
```console
$ man test | grep -A1 -- '-[sf]'
       -f FILE
              FILE exists and is a regular file
--
       -s FILE
              FILE exists and has a size greater than zero
```

The only case I see where `-s` is true but `-f` is not would be if the file was a symlink, and I don't think we need to care about excluding that.  Can we just use:

```sh
if test -s "${SHARED_DIR}/alibaba-instance-ids.txt"
```

here?  Or am I missing something?

Reply:
Sgtm, will fix.